src/main/java/be/uantwerpen/minelabs/block/entity/ChargedPointBlockEntity.java

```diff
@@ -0,0 +1,108 @@
+package be.uantwerpen.minelabs.block.entity;
+
+import be.uantwerpen.minelabs.gui.charged_point_gui.ChargedPointBlockScreenHandler;
+import be.uantwerpen.minelabs.inventory.ImplementedInventory;
+import be.uantwerpen.minelabs.util.Tags;
+import net.minecraft.block.Block;
+import net.minecraft.block.BlockState;
+import net.minecraft.entity.player.PlayerEntity;
+import net.minecraft.entity.player.PlayerInventory;
+import net.minecraft.inventory.Inventories;
+import net.minecraft.item.Item;
+import net.minecraft.item.ItemStack;
+import net.minecraft.nbt.NbtCompound;
+import net.minecraft.screen.NamedScreenHandlerFactory;
+import net.minecraft.screen.ScreenHandler;
+import net.minecraft.text.Text;
+import net.minecraft.util.collection.DefaultedList;
+import net.minecraft.util.math.BlockPos;
+import net.minecraft.util.math.Direction;
+import org.jetbrains.annotations.Nullable;
+
+import java.util.ArrayList;
+
+public class ChargedPointBlockEntity extends ChargedBlockEntity implements ImplementedInventory, NamedScreenHandlerFactory {
+
+    private final DefaultedList<ItemStack> INVENTORY = DefaultedList.ofSize(1, ItemStack.EMPTY);
+
+    public ChargedPointBlockEntity(BlockPos pos, BlockState state,
+                                   double charge,
+                                   Block anit_block,
+                                   double decay_time,
+                                   ArrayList<ItemStack> decay_drop,
+                                   Block decay_replace) {
+        super(BlockEntities.CHARGED_POINT_BLOCK_ENTITY, pos, state, charge, anit_block, decay_time, decay_drop, decay_replace);
+    }
+
+    @Override
+    public void writeNbt(NbtCompound tag) {
+        super.writeNbt(tag);
+        Inventories.writeNbt(tag, INVENTORY);
+    }
+
+    @Override
+    public void readNbt(NbtCompound tag) {
+        Inventories.readNbt(tag, INVENTORY);
+        super.readNbt(tag);
+    }
+
+    @Override
+    public DefaultedList<ItemStack> getItems() {
+        return INVENTORY;
+    }
+
+    //BLOCK HOPPER
+    @Override
+    public boolean canInsert(int slot, ItemStack stack, @Nullable Direction side) {
+        return false;
+    }
+
+    //BLOCK HOPPER
+    @Override
+    public boolean canExtract(int slot, ItemStack stack, Direction side) {
+        return false;
+    }
+
+    @Override
+    public Text getDisplayName() {
+        return Text.translatable(getCachedState().getBlock().getTranslationKey());
+    }
+
+    @Nullable
+    @Override
+    public ScreenHandler createMenu(int syncId, PlayerInventory inv, PlayerEntity player) {
+        return new ChargedPointBlockScreenHandler(syncId, inv, this);
+    }
+
+    public void setInventory(ItemStack inventoryIn) {
+        INVENTORY.set(0, inventoryIn);
+        inventoryChanged();
```

Comment:
If you use SimpleInventory instead of extending from ImplementedInventory, you can add a listener to the inventory. This makes sure the update is also called when for example the ScreenHandler edits the contents.

Reply:
The method is being used when the animation has ended and the point-charged is placed again (as an actual block, with entity).
Screenhandler already calls the onClose method of the inventory. 
Which in turn updates the field.

Reply:
Do we need to update this?

Reply:
No it's fine by me as long as it works. Ready for merge!